management_api_app/db/repositories/workspaces.py

```diff
@@ -1,43 +1,60 @@
+import uuid
 from typing import List
 
-from azure.cosmos import ContainerProxy, CosmosClient, PartitionKey
+from azure.cosmos import ContainerProxy, CosmosClient
 
 from core.config import STATE_STORE_RESOURCES_CONTAINER
-from db.errors import EntityDoesNotExist
+from db.errors import EntityDoesNotExist, UnableToAccessDatabase
 from db.repositories.base import BaseRepository
-from models.domain.resource import Resource
+from models.domain.resource import Resource, ResourceType, Status
+from models.schemas.resource import ResourceInCreate
 from resources import strings
 
 
 class WorkspaceRepository(BaseRepository):
     def __init__(self, client: CosmosClient):
-        super().__init__(client)
-        if self.database:
-            self._container = self.database.create_container_if_not_exists(id=STATE_STORE_RESOURCES_CONTAINER, partition_key=PartitionKey(path="/appId"))
-        else:
-            self._container = None
+        super().__init__(client, STATE_STORE_RESOURCES_CONTAINER)
 
     @property
     def container(self) -> ContainerProxy:
         return self._container
 
     def get_all_active_workspaces(self) -> List[Resource]:
-        workspaces = []
-        if self.container:
+        try:
             query = f'SELECT * from c ' \
                     f'WHERE c.resourceType = "{strings.RESOURCE_TYPE_WORKSPACE}" ' \
```

Comment:
IF you are always going to use isDeleted = false you should write a query builder which decorates your query string with an AND clause instead of remembering to put it everywhere

Reply:
@sachinkundu could you add a code suggestion or point to a sample

Reply:
def deleted_records(query):
      return query + 'AND deleted';
def basic_query():
      return blah
def default_query():
     return deleted_records(basic_query())

Then always use default query. Since the interface is fluent on string one could add other clauses to build a DSL. A suggestion for clarity and reducing mistakes but I'll leave it to you to think about it.

Reply:
Created #146